packages/query-core/src/utils.ts

```diff
@@ -122,10 +122,7 @@ export function matchQuery(
     return false
   }
 
-  if (
-    typeof fetchStatus !== 'undefined' &&
-    fetchStatus !== query.state.fetchStatus
-  ) {
+  if (fetchStatus && fetchStatus !== query.state.fetchStatus) {
```

Comment:
same here: `fetchStatus` is not a boolean, we just need to check if it was passed